getPythonCmd.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Search for a compatible python version on the system or in the local path
+# and return the path to it if found, otherwise returns nothing
+
+set -euo pipefail
+
+PYTHON_VERSION=${1-}
+PYTHON_LOCAL_DIR=${2-}
+PYTHON_MAJOR_VERSION=${PYTHON_VERSION%.*}
+
+if [ -z "${PYTHON_VERSION}" ]; then
+    >&2 echo "First parameter missing"
+    exit 1
+fi
+
+# shellcheck disable=SC2086
+for TMP_PYTHON_CMD in $(find ${PATH//:/ } ${PYTHON_LOCAL_DIR} -maxdepth 1 -executable -regextype sed -regex ".*/python3\.[0-9]*" 2> /dev/null | sort -V) ; do 
+    VERSION=$(${TMP_PYTHON_CMD} --version 2>&1 | awk  '{print $2}')
+    if (( $(echo "${VERSION%.*} >= ${PYTHON_MAJOR_VERSION}" | bc -l) )); then 
```

Comment:
This checks if the version is 3.7 or higher, which does not cover the case for versions such as 3.7.1 being too low. Since you have already a function that lists all executables, we could do something like this:
`PYTHON_CHECK=$(${TMP_PYTHON_CMD} import sys; expected_version = [3, 7,  4]; sys.stdout.write(str(expected_version[0] == sys.version_info.major and (expected_version[1] < sys.version_info.minor or (expected_version[1] == sys.version_info.minor and expected_version[2] <= sys.version_info.micro)))))`

Which will giveFalse or True depending on wheter the version is correct or not.

If you'd prefer the check stays entirely in Bash (which would be better for consistency, if one day python4 comes, it might change some things and make this little script crash), I would just also check for the micro version.

Reply:
I prefer leaving the check in bash and doing it entirely in bash, to avoid issue with future python4. I will also modify the script to match the exact major version (python 4.x.x would then not be used), a major version changes might not work (btw not all python2 codes works on python3). I will include the micro check.